pkg/node/node.go

```diff
@@ -348,6 +366,24 @@ func NewNode(
 			attribute.StringSlice("node_engines", executors.Keys(ctx)),
 		)
 
+		var hbClient *heartbeat.HeartbeatClient
+
+		// We want to provide a heartbeat client to the compute node if we are using NATS.
+		// We can only create a heartbeat client if we have a NATS client, and we can
+		// only do that if the configuration is available. Whilst we support libp2p this
+		// is not always the case.
+		if natsConfig != nil {
+			natsClient, err := nats_transport.CreateClient(ctx, natsConfig)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to create NATS client for node info store")
+			}
+
+			hbClient, err = heartbeat.NewClient(natsClient.Client, config.NodeID)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to create heartbeat client")
+			}
+		}
```

Comment:
Similarly, could this be done in `NewComputeNode`?

Reply:
natsConfig doesn't live any longer than NewNode(), would make more sense to do this as part of the cleanup/refactor because there's a lot of overlapping checks in NewNode() and NewCompute() doesn't really know anything about NATS (currently).